src/indicators/drought/biomasse.py

```diff
@@ -0,0 +1,335 @@
+            "DMP_MEA",
+            "DMP",
+            "biomasse_mean",
+            "biomasse",
+            "biomasse_anomaly",
+        ]
+    ]
+
+    # save file to processed filepath
+    processed_path = _get_processed_path(
+        admin=admin, iso3=None, start_dekad=start_dekad
+    )
+    df_merged.to_csv(processed_path, index=False)
+
+    return df_merged
+
+
+def load_biomasse_data(
+    admin: AdminArgument = "ADM2",
+    iso3: Union[bool, str] = None,
+    start_dekad: int = 10,
+    reprocess: bool = False,
+    redownload: bool = False,
+):
+    """Load biomasse processed data
+
+    Load the data that has been processed at the
+    specified ADM2 level and starting dekad.
+
+    Args:
+        admin: Admin level, one of 'ADM0', 'ADM1',
+            or 'ADM2'.
+        start_dekad: Starting dekad for annual
+            calculations.
+        reprocess: Boolean, whether to reprocess.
+        redownload: Boolean, whether to reprocess.
+        file_descriptor: If not None, file descriptor
+            to access non-admin file.
+
+    Returns:
+        Pandas data frame.
+    """
+    if reprocess:
+        return process_dmp(
+            admin=admin, start_dekad=start_dekad, redownload=redownload
+        )
+    else:
+        _check_admin(admin)
+        processed_path = _get_processed_path(
+            admin=admin, iso3=iso3, start_dekad=start_dekad
+        )
+        return pd.read_csv(processed_path)
+
+
+def aggregate_biomasse(
+    admin_pcodes: List[str],
+    admin: AdminArgument = "ADM2",
+    start_dekad: int = 10,
+    iso3: Union[bool, str] = None,
+):
+    """Aggregate biomasse data to set of areas
+
+    While the generic processed administrative
+    biomasse data is useful, sometimes we want
+    to aggregate the biomasse values to a set
+    of administrative areas. This should be done
+    by summing up the mean biomasse for each dekad
+    for those areas and the reported biomasse for
+    each year and dekad, then recalculating the
+    anomaly.
+    """
+    _check_admin(admin)
+    df = load_biomasse_data(admin=admin, start_dekad=start_dekad)
+    admin_col = admin.replace("ADM", "admin") + "Pcod"
+    df_subset = df[df[admin_col].isin(admin_pcodes)]
+    # just keep to year 2000 for unique value per dekad
+    # when calculating biomasse mean aggregated
+    df_mean = df_subset[df.year.isin([2000])]
+    df_mean = (
+        df_mean[["dekad", "biomasse_mean"]]
+        .groupby("dekad")
+        .sum()
+        .reset_index()
+    )
+    df_dekad = (
+        df_subset[["year", "dekad", "biomasse"]]
+        .groupby(["year", "dekad"])
+        .sum()
+        .reset_index()
+    )
+    df_merged = pd.merge(df_dekad, df_mean, on=["dekad"])
+    df_merged["biomasse_anomaly"] = (
+        100 * df_merged["biomasse"] / df_merged["biomasse_mean"]
+    )
+
+    # sorting data again year then dekad
+    df_merged.sort_values(by=["year", "dekad"], inplace=True)
+
+    if iso3 is not None:
```

Comment:
why would you only save it if you have an `iso3`? Maybe in this case do not allow the `iso3` to be `None`?

Reply:
In this case @Tinkaa, my thought process was you might want to use the function without saving automatically to a specific country file. Maybe if you were exploring for instance cross-country analysis? Then you would need to manually save because we don't have an immediately logical filename and folder in our structure for it to fit into. What do you think? Previously I actually had it as a filepath argument, but realized in general we are wanting to save to an `iso3` folder.

Reply:
Hmm yeah I have been thinking about this as well.. I figured that if the case occurs of cross-country analysis we might as well come up with a fake iso3. So basically that the iso3 arg just serves as an ID. Though that might get a bit ambiguous, especially if we want to check for validity of the iso3 in the toolbox. 

It is just that if we add the None option for iso3 we would basically have to check for that with every dataset that covers more than a country before saving. Which might be a bit ugly. 

However, for now it is fine to leave it in. The thing I would change though then is to make the iso3 a mandatory argument instead of having the default as None. But then keep the option for now to be None instead of a str. Would that make sense? 

@turnerm please chime in if you got any thoughts on this topic beyond this specific case.

Reply:
I like that idea, let's see what @turnerm thinks.